db/commitlog/commitlog.cc

```diff
@@ -1881,6 +1909,9 @@ void db::commitlog::segment_manager::on_timer() {
         if (cfg.mode != sync_mode::BATCH) {
             sync();
         }
+
+        _sync_semaphore.signal(_sync_semaphore.waiters());
```

Comment:
I don't understand how you can release _sync_semaphore without waiting on some from sync().

Reply:
Depends on whether we assume this is signalled when sync is initiated or done. But I see your point. Can change it to be signalled on sync completion.

Reply:
Done